### PR TITLE
#123 [Azure] Event Hub - capacity detector

### PR DESCRIPTION
**Pull request by Shr3ps**

### CODE REVIEW

File: modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  B:
+    metric: IncomingRequests
+  signal:
+    formula: "(A/B).scale(100).fill(0)"
+rules:
+  critical:
+    threshold: 90
+    comparator: ">"
+  major:
+    threshold: 80
```
Comment (BzSpi) [inline modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml:20]:
```suggestion
    threshold: 0
```
We should be warned if we have any throttled request.

File: modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+signals:
+  A:
+    metric: ThrottledRequests
+  B:
+    metric: IncomingRequests
+  signal:
+    formula: "(A/B).scale(100).fill(0)"
+rules:
+  critical:
+    threshold: 90
```
Comment (BzSpi) [inline modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml:17]:
```suggestion
    threshold: 10
```
Not sure if 10% is a good threshold but 90% is way to high.

File: modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+---
+module: "Azure Event Hub"
+name: "throttled requests"
+filtering: "filter('resource_type', 'Microsoft.EventHub/clusters') and filter('primary_aggregation_type', 'true')"
+aggregation: ".mean(by=['azure_resource_name', 'azure_resource_group_name', 'azure_region'])"
+transformation: ".max(over='15m')"
```
Comment (BzSpi) [inline modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml:7]:
```suggestion
transformation: ".min(over='15m')"
```
To avoid any spike.

**Review by BzSpi — CHANGES_REQUESTED** [review]

**Comment by xp-1000** [comment]
please see https://github.com/claranet/terraform-signalfx-detectors/pull/217 for a fix on deadlink

File: modules/integration_azure-event-hub/common-modules.tf [hunk 4]
```diff
@@ -0,0 +1 @@
+../../common/module/modules.tf
```
Comment (BzSpi) [inline modules/integration_azure-event-hub/common-modules.tf:1]: The common module use `env` and `sfx_monitored` filter but we want `azure_tag_env` and `azure_tag_sfx_monitored`.

File: modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    comparator: ">"
+  major:
+    threshold: 30
+    comparator: ">"
+    dependency: critical
+  minor:
+    threshold: 10
+    comparator: ">"
+    dependency: major
+  warning:
```
Comment (xp-1000) [inline modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml:28]: is it really useful / desirable to have such a lot of severities ?

the role of this repo is also to advice a relevant severity for each detector created following our https://github.com/claranet/terraform-signalfx-detectors/wiki/Notifications-binding.

**Review by xp-1000 — CHANGES_REQUESTED** [review]

File: modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    threshold: 1
```
Comment (BzSpi) [inline modules/integration_azure-event-hub/conf/01-capacity-throttling.yaml:29]: With a strict comparison, we can set 0 as value.

**Review by xp-1000 — APPROVED** [review]
LGTM

**Pull request merged**